fabric-ca-client/lib/FabricCAServices.js

```diff
@@ -77,7 +71,9 @@ const FabricCAServices = class extends BaseClient {
 			this.setCryptoSuite(_cryptoSuite);
 		} else {
 			this.setCryptoSuite(utils.newCryptoSuite());
-			this.getCryptoSuite().setCryptoKeyStore(utils.newCryptoKeyStore());
+			const kvs = new InMemoryKeyValueStore();
+			const cks = new CryptoKeyStore(kvs);
```

Comment:
I guess the key-value store and CryptoKeyStore are being created explicitly here rather than calling `utils.newCryptoKeyStore()` because it needs to be done synchronously within a constructor. This means that `await cks.initialize()` doesn't get called as it normally would, but this is OK because we know that all that call does is call `await kvs.initialize()` and initialize is a no-op on InMemoryKeyValueStore. That is a lot of information required to see that this code is correct, and not at all obvious. This makes me very nervous about maintainability!

Reply:
To make it easy and maintainable, I think we could even skipping this part. So that we does not allow undefined cryptoSuite as parameter, developer have to provide it explicitly meanwhile ensure the cryptoSuite is read to use (with an initialized cryptoKeystore) . We will never assign a default implementation to them